workflow/scripts/get-transcript-info.R

```diff
@@ -39,69 +40,124 @@ while ( class(mart)[[1]] != "Mart" ) {
       }
       # hop to next mirror
       mart <- switch(mart,
-                     useast = "uswest",
-                     uswest = "asia",
-                     asia = "www",
-                     www = {
-                       # wait before starting another round through the mirrors,
-                       # hoping that intermittent problems disappear
-                       Sys.sleep(30)
-                       "useast"
-                     }
-              )
+        useast = "uswest",
+        uswest = "asia",
+        asia = "www",
+        www = {
+          # wait before starting another round through the mirrors,
+          # hoping that intermittent problems disappear
+          Sys.sleep(30)
+          "useast"
+        }
+      )
     }
   )
 }
+three_prime_activated <- snakemake@params[["three_prime_activated"]]
 
 attributes <- c("ensembl_transcript_id",
                 "ensembl_gene_id",
                 "external_gene_name",
                 "description")
-
-has_canonical <- "transcript_is_canonical" %in% biomaRt::listAttributes(mart=mart)$name
+has_canonical <-
+  "transcript_is_canonical" %in% biomaRt::listAttributes(mart = mart)$name
 
 if (has_canonical) {
   attributes <- c(attributes, "transcript_is_canonical")
 }
 
-t2g <- biomaRt::getBM(
-            attributes = attributes,
-            mart = mart,
-            useCache = FALSE
-            )
 if (!has_canonical) {
   t2g <- t2g %>% add_column(transcript_is_canonical = NA)
```

Comment:
This will fail in case has_canonical is false, because t2g is not created above.

Reply:
Fixed the case